src/app/cart-item/cart-item.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Item } from '../models/item';
+import { CartService } from '../services/cart.service';
+import { ImageService } from '../utilities/generateImagePath.service.';
+
+@Component({
+  selector: 'app-cart-item',
+  templateUrl: './cart-item.component.html',
+  styleUrls: ['./cart-item.component.css'],
+})
+export class CartItemComponent implements OnInit {
+  @Input('cartItem') cartItem?: [Item, number];
```

Comment:
This would be easier to work with if you had created a `CartItem` type, and put the quantity on it. 

Something like this:

```
interface CartItem {
    item: Item
    quantity: number
}
```

Then, you could have used this type as your Input

Reply:
Thanks for advice! I always have problem how to store data easily